7 - 12: Add exercises and solutions (#31)
ONSdigital/r-rap-demo: exercises/solutions/03_config_files_solutions.Rmd

# Solutions: Using configuration files in your RAP pipeline

This notebook provides step-by-step solutions for adding and using configuration parameters in a RAP pipeline. Each step matches the corresponding exercise notebook and demonstrates a modular approach.

```{r}
# Setup code for solutions

library(yaml)

> source("D:/repos/R_rap_demo/R/R_rap_demo/cleaning.R")

Reviewer: Just to highlight, will need to remove the `R_rap_demo` text in the solutions files and the filepaths should be relative so don't need "D:/repos/R_rap_demo/"

source("D:/repos/R_rap_demo/R/R_rap_demo/processing.R")
source("D:/repos/R_rap_demo/R/R_rap_demo/report.R")
source("D:/repos/R_rap_demo/R/R_rap_demo/utils.R")

# Construct the path to the YAML configuration file
config_path <- file.path("D:/repos/R_rap_demo/config/user_config.yaml")

# Load the YAML configuration file
config <- yaml::read_yaml(config_path)

input_path = config$input_path

# I/O Read data
health_df <- read_health_data(input_path)

# Output path
output_path <- "D:/repos/R_rap_demo/data/outputs/cleaned/health_data_cleaned"
```

## Step 1 Solution: Add a new parameter to the config file

Open `config/user_config.yaml` and add the following parameter:

```         
min_height_cm: 155  # Minimum height (cm) to include in analysis
```

This parameter allows you to control which rows are included based on height. Save the file after editing.

## Step 2 Solution: Read config parameters in a modular way

Create a function to read config values in a new module, e.g., `R/R_rap_demo/io.R`:

```{r}
# R/main.R

min_height <- config$min_height
```

```{r}
print(min_height)
```

## Step 3 Solution: Use config parameters in your pipeline

> Update your pipeline (e.g., in `main.R` or a cleaning module) to use the config-driven parameter for filtering:

Reviewer: ```suggestion
Update your pipeline (e.g., in `main.R` and the cleaning.R module) to use the config-driven parameter for filtering:
```

Same for python
